spreadsheets.js

```diff
@@ -148,6 +119,101 @@ ScheduleParsingError.prototype = new Error();
 function PairingError () {}
 PairingError.prototype = new Error();
 
+// Get an appropriate set of base tokens for the scheduling messages
+function get_tokens_scheduling(input_string){
+    // Do some basic preprocessing
+    // Replace non-date punctuation/symbols with spaces
+    input_string = input_string.replace(/[@,\(\)]/g, ' ');
+    input_string = input_string.replace(/[<\>]/g, '');
+
+    var parts = _.map(input_string.split(" "), function(item) {
+        // Remove . and / and : and - from the beginning and end of the word
+        item = item.replace(/^[:\.\/-]/g, '');
+        item = item.replace(/[:\.\/-]$/g, '');
+
+        return item;
+    });
+    parts = _.filter(parts, function(i) { return i.length > 0; });
+    parts = _.filter(parts, function(i) {
+        return WORDS_TO_IGNORE.indexOf(i.toLowerCase()) == -1;
+    });
+    return parts;
+}
+
+// Take the string they posted and turn it into a set of possible date
+// strings by doing things like:
+//
+// Unifying the punctuation.
+// removing gmt/utc
+// Adding the current-year
+// replacing day of the week tokens with the appropriate date format
+function get_possible_date_strings(date_string, extrema) {
+    var date_strings = [];
+
+    // Unify some common stuff first.
+    //
+    // Change /.: to -
+    date_string = date_string.replace(/[\/:\.]/g, '-');
+    date_string = date_string.toLowerCase();
+    date_string = date_string.replace(/gmt/g, "");
+    date_string = date_string.replace(/utc/g, "");
+
+    var date_name_mappings = {};
+
+    var cur = extrema.start.clone();
+    var now = moment.utc();
+    while (!cur.isAfter(extrema.end)) {
+        var month_day = cur.format("MM-DD");
+        // Deal with a couple of formats that moment doesn't produce but are used.
+        if (cur.format("dddd") == "Thursday") {
+            date_name_mappings["thurs"] = month_day;
```

Comment:
I don't understand why we have a special case for Thurs and Weds... You mean those are not supported abbreviations?